src-tauri/src/plugins/database.rs

```diff
@@ -573,8 +576,18 @@ pub fn init<R: Runtime>(db: DB) -> TauriPlugin<R> {
             reset,
             import_tracks_to_library,
         ])
-        .setup(|app_handle, _api| {
-            app_handle.manage(db);
+        .setup(move |app_handle, _api| {
+            let app_handle = app_handle.clone();
+            tauri::async_runtime::spawn(async move {
+                let db = match setup(&app_handle).await {
+                    Ok(db) => db,
+                    Err(err) => {
+                        error!("Failed to setup database: {:?}", err);
+                        return;
+                    }
+                };
+                app_handle.manage(db);
+            });
```

Comment:
very nice!

Reply:
Not sure about the timing implications. The plugin state might become available after the setup stage has already completed.

But it seems to work as expected, is supported according to the thread-safety constraints and I don't know of any alternative solutions.